src/main/kotlin/no/nav/syfo/personoppgave/api/v2/VeilederPersonOppgaveApiV2.kt

```diff
@@ -0,0 +1,97 @@
+package no.nav.syfo.personoppgave.api.v2
+
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import no.nav.syfo.auth.getNAVIdentFromToken
+import no.nav.syfo.client.veiledertilgang.VeilederTilgangskontrollClient
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.personoppgave.PersonOppgaveService
+import no.nav.syfo.personoppgave.domain.toPersonOppgaveVeileder
+import no.nav.syfo.util.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.util.*
+
+private val log: Logger = LoggerFactory.getLogger("no.nav.syfo")
+
+const val registerVeilederPersonOppgaveApiV2BasePath = "/api/v2/personoppgave"
+
+fun Route.registerVeilederPersonOppgaveApiV2(
+    personOppgaveService: PersonOppgaveService,
+    veilederTilgangskontrollClient: VeilederTilgangskontrollClient,
+) {
+    route(registerVeilederPersonOppgaveApiV2BasePath) {
+        get("/personident") {
+            try {
+                val callId = getCallId()
+                val token = getBearerHeader()
+                    ?: throw IllegalArgumentException("No Authorization header supplied")
+
+                val personIdent = call.request.headers[NAV_PERSONIDENT_HEADER.toLowerCase()]
+                    ?: throw IllegalArgumentException("No PersonIdent supplied")
+                val fnr = PersonIdentNumber(personIdent)
+
+                when (veilederTilgangskontrollClient.hasAccessWithOBO(fnr, token, callId)) {
```

Comment:
like greit med if / else her i stedet for when / true / else kanskje?

Reply:
Jepp, har erstattet When med if/else nå.